### PR TITLE
Deprecated `expired` and `credentialsExpired`

### PR DESCRIPTION
**Pull request by XWB**
Replaces https://github.com/FriendsOfSymfony/FOSUserBundle/pull/1751

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ Changelog
 * [BC break] The templating engine configuration has been removed, as well as the related code.
 * [BC break] Changed the XML namespace to `http://friendsofsymfony.github.io/schema/dic/user`
 * [BC break] Added `UserInterface::getId`.
+* [BC break] Removed unused properties `expired` and `credentialsExpired` including corresponding methods. This may break code,
+  makes use of this methods, extending classes, and/or existing installations because of missing mappings for required db fields.
```
Comment (stof) [inline Changelog.md]: this sentence does not make sense. Looks like some words are missing

File: Model/User.php [hunk 2]
```diff
@@ -190,9 +176,7 @@ public function unserialize($serialized)
             $this->salt,
             $this->usernameCanonical,
             $this->username,
-            $this->expired,
             $this->locked,
-            $this->credentialsExpired,
```
Comment (stof) [inline Model/User.php]: This breaks BC in a very bad way: unserializing data from the old version of the bundle would create a corrupted object now (storing things in the wrong properties), and the user is serialized in the session. Until now, we always worked hard on preserving BC here.

**Comment by stof** [comment]
@XWB please don't merge PRs so fast after opening it. It makes the review impossible. And the issue in your implementation was not present in the original PR.

**Comment by stof** [comment]
Btw, the title of the PR is wrong too. This does not deprecate things, it deletes them. A deprecated API is still there, and does not break anything.

**Comment by stof** [comment]
@XWB I will revert this merge so that few people will upgrade their project to the version breaking unserialization (as fixing it would break it again for them). Such change should be submitted again in a non-broken way.

**Comment by KingCrunch** [comment]
Hi @XWB 

I'm just curious, but is there's a reason you copied my PR ( #1751 ) including all the bad language? Currently the only difference I can see is, that you introduced a BC-break, whereas mine didn't. I'm seriously a little bit upset about this behaviour...

**Comment by gondo** [comment]
I'am with @KingCrunch here, why was the existing ticket recreated?

**Pull request merged**